### ISSUE TITLE
Help with webpage interface...

### ISSUE THREAD

**Issue by doublee313** [knightzac19/NodeJS-ArkService#3]
Zac,
I know you have been offline Discord, but I made some progress and got the NodeJS installed. I have everything working, just need some help with some example html pages, etc. My end goal is to have a nice interface for my players to goto but I just needed some examples so I can play with the code. I'm not a great web programmer, so I am learning. Thank you.

**Comment by Proantagonist** [comment]
I'll provide some help, given this utility has been amazing for my purposes. 

Examples will very much depend on the scope of your project including the framework, if any.

The easiest example he provides is in the form of a JQuery post request as shown on the Readme:

`
jQuery.post({url:"http://localhost:8081/listOnline",data:JSON.stringify({api_key:"YOUR_KEY"}),success:function(d){console.log(d);},contentType:"application/json"});
`

Output
`
{
	players: ["","first_player","second_player",""]
}
`

Once you are able to verify that you are able to connect to and receive an object response, you can use the model to address where and how you want to display the data.

For testing purposes, or in the simplest form, you can use do:

`$(body).append(d.players);`

Put that in the post callback and it will print out the player array.

Hope this helped. Due to the number of frameworks and implementations, it's difficult to give a concrete example.

**Comment by doublee313** [comment]
Thanks Edgar! This will be very helpful for me. If I can request one thing it will make my life easier to get this project going.

First off, I’m a network engineer and have limited coding experience. I did start some PHP and HTML back in 99-01. Then I have just played around with piecing code together and got by. Lol. So I just need to see an example that works then I can dissect it and get my results.

I have ARKNodeJS running now and it can be accessed from the internet. My issue is I need a webpage that works to reference my ARKData. Without me knowing how to put this code into a PHP/Java//HTML page, I’m sort of lost. Here’s what I’m looking to do and really been trying hard with the PHP version of this but I would rather put my time into this one as he’s updating it. Here’s the page I want to reproduce (he wont share his code):  http://amityisland.info/survivors.php  I’m willing to share mine when it’s complete as I like helping others. But I understand some people like to keep their work private (I understand). 

That’s really all I needed to start. You info below I understand, but making it work in a webpage is my issue (calling .cs files, etc). 

Any help would be greatly appreciated. 

Thank you,
Eddie

**Comment by Proantagonist** [comment]
You don't really need to call any .cs pages or have any backend apart from the NodeJS service. Everything else can be done with front-end technologies because the NodeJS service provides API endpoints which return a valid javascript object containing the information requested. For example, using the examples outlined in the README, have a JS file with all your service calls and reference that js file into the html page. That really is as simple as you can make this work.

To recap:

index.html // markup which determines layout
 |----> main.js // front-end logic. Has the REST calls to the NodeJS service, and DOM manipulations
 |----> styles.css // CSS styles used to theme your layout

Granted, this isn't the BEST way to do it, but it will work. Let me know if this clarifies things for you.
